syntax = "proto3";


package PublishSubscribe;

//import "google/protobuf/empty.proto";


//service NotificationObserver {
//	rpc Notify(stream Notification) returns (google.protobuf.Empty) {}
//}

service NotificationSubscriber {
	rpc Subscribe(NotificationChannel) returns (stream Notification) {}
}

message Notification {
	uint64 index = 1;
	string content = 2;
}

message NotificationChannel {
	string id = 1;
}
